### PR TITLE
Allow to override generated reports base directory.

### LINKED ISSUE #321: Exposing output file for reporters in plugin ktlint properties
We would like to have the option to choose the location for reporter, which in ktlint is optional, but in plugin it's hardcoded and not available for setting.

```
        project.layout.buildDirectory.file(project.provider {
            "reports/ktlint/${this@BaseKtlintCheckTask.name}.$fileExtension"
        })
```

### PR DESCRIPTION
**Pull request by Tapchicoma**
Closes #321

### CODE REVIEW

File: plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/BaseKtlintCheckTask.kt [hunk 1]
```diff
@@ -32,13 +33,34 @@ import org.jlleitschuh.gradle.ktlint.reporter.ReporterType
...
+     */
+    @get:Internal
+    val reporterOutputDir: DirectoryProperty = objectFactory.directoryProperty().convention(
+        project.layout.buildDirectory.dir("reports/ktlint")
+    )
+
+    /**
+     * [DirectoryProperty] takes into account also content of directory, when snapshotting input.
+     * As reports generated are always generated in this directory, this prevents task proper caching,
+     * so this input exists as a workaround for the described problem.
```
Comment (JLLeitschuh) [inline plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/BaseKtlintCheckTask.kt]: Can you elaborate? I'm missing something.

Comment (JLLeitschuh) [reply plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/BaseKtlintCheckTask.kt]: Task caching also snapshots the outputs. If the output has changed, the task is marked as out of date. Why do we need this?

Comment (Tapchicoma) [reply plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/BaseKtlintCheckTask.kt]: You are right, was confused by https://docs.gradle.org/current/javadoc/org/gradle/api/tasks/InputDirectory.html. Actually `reporterOutputDir` could be just annotated with `@get:Input`.

Comment (Tapchicoma) [reply plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/BaseKtlintCheckTask.kt]: Force updated and removed this property.

Comment (Tapchicoma) [reply plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/BaseKtlintCheckTask.kt]: or not... running tests  in IDEA was green, though from command line they fail :thinking:

Comment (Tapchicoma) [reply plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/BaseKtlintCheckTask.kt]: Should be fine now

File: README.md [hunk 2]
```diff
@@ -282,6 +283,32 @@ dependencies {
...
 
+#### Setting reports output directory
+
+It is possible also to define different from default output directory for generated reports
+ (by default it is "build/reports/ktlint"):
+<details>
+<summary>Groovy</summary>
+
+```groovy
+tasks.withType(org.jlleitschuh.gradle.ktlint.KtlintCheckTask.class) {
```
Comment (JLLeitschuh) [inline README.md:294]:
```suggestion
tasks.withType(org.jlleitschuh.gradle.ktlint.KtlintCheckTask) {
```

In groovy, I don't think you need the `.class`

**Pull request merged**